pkg/scheduler/podtask/pod_task.go

```diff
@@ -183,6 +184,98 @@ func New(ctx api.Context, pod *api.Pod, executor *mesos.ExecutorInfo) (*T, error
 	return task, nil
 }
 
+func (t *T) SaveRecoveryInfo(dict map[string]string) {
+	dict[annotation.TaskIdKey] = t.ID
+	dict[annotation.SlaveIdKey] = t.TaskInfo.SlaveId.GetValue()
+	dict[annotation.OfferIdKey] = t.Offer.Details().Id.GetValue()
+	dict[annotation.ExecutorIdKey] = t.TaskInfo.Executor.ExecutorId.GetValue()
+}
+
+// reconstruct a task from metadata stashed in a pod entry. there are limited pod states that
+// support reconstruction. if we expect to be able to reconstruct state but encounter errors
+// in the process then those errors are returned. if the pod is in a seemingly valid state but
+// otherwise does not support task reconstruction return false. if we're able to reconstruct
+// state then return a reconstructed task and true.
+//
+// at this time task reconstruction is only supported for pods that have been annotated with
+// binding metadata, which implies that they've previously been associated with a task and
+// that mesos knows about it.
+//
+func RecoverFrom(pod *api.Pod) (*T, bool, error) {
+	if pod == nil {
+		return nil, false, fmt.Errorf("illegal argument: pod was nil")
+	}
+	if pod.Status.Host == "" || len(pod.Annotations) == 0 {
```

Comment:
If the binding fails, then Host == "" but pod status == Failed, so we can actually check for that here.